src/Sarif.Sarifer/BackgroundAnalysisTextViewCreationListener.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel.Composition;
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    /// <summary>
+    /// An <see cref="ITextViewCreationListener"/> that triggers background analysis.
+    /// </summary>
+    [ContentType(AnyContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    [Export(typeof(ITextViewCreationListener))]
+    public class BackgroundAnalysisTextViewCreationListener : ITextViewCreationListener
+    {
+        private const string AnyContentType = "any";
+
+#pragma warning disable CS0649 // Filled in by MEF
+#pragma warning disable IDE0044 // Assigned by MEF
+        [Import]
+        private Lazy<IBackgroundAnalysisService> backgroundAnalysisService;
+
+        [Import]
+        private Lazy<IVsEditorAdaptersFactoryService> vsEditorAdaptersFactoryService;
+#pragma warning restore IDE0044
+#pragma warning restore CS0649
+
+        /// <inheritdoc/>
+        public void TextViewCreated(ITextView textView)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            textView = textView ?? throw new ArgumentNullException(nameof(textView));
+
+            string text = textView.TextBuffer.CurrentSnapshot.GetText();
+            string path = GetPathFromTextView(textView);
+
+            this.backgroundAnalysisService.Value.StartAnalysis(path, text);
```

Comment:
>StartAnalysis [](start = 49, length = 13)

So when the text view is closed, do we want to stop analysis? (Should this be async and take a cancellation token?)

Reply:
Absolutely. Future work based on this PR as a starting point.

---
In reply to: [519924051](https://github.com/microsoft/sarif-visualstudio-extension/pull/260#discussion_r519924051) [](ancestors = 519924051)